girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -216,11 +218,21 @@ def finalizeUpload(self, upload, file):
         path = os.path.join(dir, hash)
         abspath = os.path.join(self.assetstore['root'], path)
 
+        # Store the hash in the upload so that deleting a file won't delete
+        # this file
+        upload['sha512'] = hash
+        upload = self.model('upload').save(upload, validate=False, triggerEvents=False)
+
         mkdir(absdir)
 
-        if os.path.exists(abspath):
+        # Only maintain the lock which checking if the file exists.  The only
+        # other place the lock is used is checking if an upload task has
+        # reserved the file, so this is sufficient.
+        with self.deleteLock:
+            pathExists = os.path.exists(abspath)
+        if pathExists:
             # Already have this file stored, just delete temp file.
-            os.remove(upload['tempFile'])
+            os.unlink(upload['tempFile'])
```

Comment:
We use `unlink` in some spots in Girder and `remove` in others.  I think we should be consistent, even though they are the same thing.  It looks like we mildly prefer `unlink`, so I figured I would change this as a move for more consistency.